contrib/neml2.mk

```diff
@@ -0,0 +1,41 @@
+ifeq ($(ENABLE_LIBTORCH),true)
+NEML2_DIR            ?= $(BLACKBEAR_DIR)/contrib/neml2
+
+ifneq ($(wildcard $(NEML2_DIR)/CMakeLists.txt),)
+
+NEML2_INCLUDE        := $(NEML2_DIR)/include
+NEML2_SRC            := $(shell find $(NEML2_DIR)/src -name "*.cxx")
+NEML2_OBJ            := $(patsubst %.cxx,%.$(obj-suffix),$(NEML2_SRC))
+NEML2_LIB            := $(NEML2_DIR)/libNEML2-$(METHOD).la
+
+$(APPLICATION_DIR)/lib/libblackbear-$(METHOD).la: $(NEML2_LIB)
+
+$(NEML2_LIB): $(NEML2_OBJ)
+	@echo "Linking Library "$@"..."
+	@$(libmesh_LIBTOOL) --tag=CC $(LIBTOOLFLAGS) --mode=link --quiet \
+	  $(libmesh_CC) $(libmesh_CFLAGS) -o $@ $(NEML2_OBJ) $(libmesh_LDFLAGS) $(EXTERNAL_FLAGS) -rpath $(NEML2_DIR)
+	@$(libmesh_LIBTOOL) --mode=install --quiet install -c $(NEML2_LIB) $(NEML2_DIR)
+
+$(NEML2_DIR)/src/%.$(obj-suffix) : $(NEML2_DIR)/src/%.cxx
+	@echo "Compiling C++ (in "$(METHOD)" mode) "$<"..."
+	@$(libmesh_LIBTOOL) --tag=CXX $(LIBTOOLFLAGS) --mode=compile --quiet \
+	  $(libmesh_CXX) $(libmesh_CPPFLAGS) $(ADDITIONAL_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) -w -DHAVE_CONFIG_H -MMD -MP -MF $@.d -MT $@ -c $< -o $@
+
+ADDITIONAL_INCLUDES  += -iquote$(NEML2_INCLUDE)
+ADDITIONAL_LIBS      += -L$(NEML2_DIR) -lNEML2-$(METHOD) -lstdc++fs
+ADDITIONAL_CPPFLAGS  += -DNEML2_ENABLED -DDTYPE=Float64
+NONUNITY_DIRS        += $(shell find src/nonunity -type d -not -path '*/.libs*' 2> /dev/null)
+NONUNITY_DIRS        += $(shell find test/src/nonunity -type d -not -path '*/.libs*' 2> /dev/null)
+app_non_unity_dirs   += $(foreach i, $(NONUNITY_DIRS), %$(i))
```

Comment:
This is because of namespace clash between moose and libtorch, citing the moose doc:

> Due to namespace conflicts between MOOSE and libtorch, the folders of source files containing libtorch includes need to be excluded from the unit build directories. The makefile of the Stochastic Tools Module serves as a good example on how to achieve this for applications.

I believe some work is scheduled to fix the namespace clash in this FY, so I think the naming convention for these folders doesn't matter too much here given that they will go away soon.

Reply:
Thanks, good to know the reason for that. I agree then that we should leave as-is.